zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/DeploymentReconstructProcessor.java

```diff
@@ -139,20 +139,15 @@ private DeploymentRecord recreateDeploymentForResources(
     deploymentRecord.setDeploymentKey(deploymentKey);
     deploymentRecord.setTenantId(tenantId);
     for (final var resource : resources) {
-      attachResourceToDeployment(deploymentRecord, resource);
+      attachResourceMetadataToDeployment(deploymentRecord, resource);
     }
     return deploymentRecord;
   }
 
-  private void attachResourceToDeployment(
+  private void attachResourceMetadataToDeployment(
```

Comment:
❓ I assume when scaling you will iterate over the deployments, grab the binary from the state using the resource metadata available and add them to the record before distributing?

Reply:
Exactly. No need to have the binary resource duplicated if we can also just look it up when needed.